src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -249,6 +249,8 @@ mongoc_client_pool_pop (mongoc_client_pool_t *pool)
 {
    mongoc_client_t *client;
    int32_t wait_queue_timeout_ms;
+   int32_t expire_at_ms = -1;
+   int32_t now_ms;
```

Comment:
`bson_get_monotonic_time()` returns an int64_t (and it does not necessarily correspond to the microseconds since epoch) I don't think we can be assured that it fits in an int32_t, even after converting to milliseconds. I'd suggest declaring these as int64_t, and casting the difference as int32_t when passing to `mongoc_cond_timedwait`, which should fit in an int32_t.

As a sidenote, I recently had an overflow issue while working on CDRIVER-3535 caused by me mishandling the result of bson_get_monotonic_time in int32_t.

Reply:
It turns out that `mongoc_cond_timedwait` takes and `int64_t`, so I am not going to cast.  I'll just change the types.